### PR TITLE
add docs

### LINKED ISSUE #4: improve docs

### LINKED ISSUE #6: getBinaries() ==> getBinScripts()?

### PR DESCRIPTION
**Pull request by devoncarew**
- fix #4 
- fix #6 
- add docs, and rename a method

@pq @keertip

### CODE REVIEW

File: lib/pub_cache.dart [hunk 1]
```diff
@@ -10,69 +10,84 @@ import 'package:path/path.dart' as path;
...
+    if (refs.length == 1) return refs.first;
+
+    PackageRef latest = refs.first;
+
+    if (includePreRelease) {
+      for (int i = 1; i < refs.length; i++) {
+        if (refs[i].version > latest.version) latest = refs[i];
+      }
+    } else {
+      List<Version> versions = refs.map((ref) => ref.version).toList();
```
Comment (pq) [inline lib/pub_cache.dart:81]: Interesting that you need to convert to a list here.  I notice that semver `Version.primary` is expecting one when an `Iterable` would do.  But  anyway!

File: lib/pub_cache.dart [hunk 2]
```diff
@@ -140,11 +167,18 @@ class Application {
...
+  /// The name of the package.
   String get name;
+  /// The version of the package.
   Version get version;
 
+  /// Resolve the package reference into the actual package, including the
+  /// location on disk.
   Package resolve();
 
   bool operator ==(other) {
```
Comment (pq) [inline lib/pub_cache.dart:184]: Does this deserve an `@override` annotation?

**Comment by pq** [comment]
LGTM!

File: lib/pub_cache.dart (hunk 2 above)
Comment (devoncarew) [inline lib/pub_cache.dart:184]: Hmm, I generally don't use them. Not sure how common it is -

**Pull request merged**